frontend/src/api/k8s/__tests__/inferenceServices.spec.ts

```diff
@@ -1,26 +1,65 @@
+import {
+  K8sStatus,
+  k8sCreateResource,
+  k8sDeleteResource,
+  k8sGetResource,
+  k8sListResource,
+  k8sUpdateResource,
+} from '@openshift/dynamic-plugin-sdk-utils';
 import { mockAcceleratorProfile } from '~/__mocks__/mockAcceleratorProfile';
+import { mockInferenceServiceK8sResource } from '~/__mocks__/mockInferenceServiceK8sResource';
 import { mockInferenceServiceModalData } from '~/__mocks__/mockInferenceServiceModalData';
-import { assembleInferenceService } from '~/api/k8s/inferenceServices';
+import { mockK8sResourceList } from '~/__mocks__/mockK8sResourceList';
+import { mock200Status, mock404Error } from '~/__mocks__/mockK8sStatus';
+import { mockProjectK8sResource } from '~/__mocks__/mockProjectK8sResource';
+import {
+  assembleInferenceService,
+  createInferenceService,
+  deleteInferenceService,
+  getInferenceService,
+  getInferenceServiceContext,
+  listInferenceService,
+  listScopedInferenceService,
+  updateInferenceService,
+} from '~/api/k8s/inferenceServices';
+import { InferenceServiceModel } from '~/api/models';
+import { InferenceServiceKind, ProjectKind } from '~/k8sTypes';
 import { translateDisplayNameForK8s } from '~/pages/projects/utils';
 import { AcceleratorProfileState } from '~/utilities/useAcceleratorProfileState';
 
+jest.mock('@openshift/dynamic-plugin-sdk-utils', () => ({
+  k8sListResource: jest.fn(),
+  k8sGetResource: jest.fn(),
+  k8sCreateResource: jest.fn(),
+  k8sUpdateResource: jest.fn(),
+  k8sDeleteResource: jest.fn(),
+}));
+
+const k8sListResourceMock = jest.mocked(k8sListResource<InferenceServiceKind | ProjectKind>);
+const k8sGetResourceMock = jest.mocked(k8sGetResource<InferenceServiceKind>);
+const k8sCreateResourceMock = jest.mocked(k8sCreateResource<InferenceServiceKind>);
+const k8sUpdateResourceMock = jest.mocked(k8sUpdateResource<InferenceServiceKind>);
+const k8sDeleteResourceMock = jest.mocked(k8sDeleteResource<InferenceServiceKind, K8sStatus>);
+
 global.structuredClone = (val: unknown) => JSON.parse(JSON.stringify(val));
 
 describe('assembleInferenceService', () => {
   it('should have the right annotations when creating for Kserve', async () => {
     const inferenceService = assembleInferenceService(mockInferenceServiceModalData({}));
 
     expect(inferenceService.metadata.annotations).toBeDefined();
-    expect(inferenceService.metadata.annotations?.['serving.kserve.io/deploymentMode']).toBe(
-      undefined,
-    );
+    expect(
+      inferenceService.metadata.annotations?.['serving.kserve.io/deploymentMode'],
+    ).toStrictEqual(undefined);
```

Comment:
Using `toBe` is correct for primitive type assertions or any other time identity equality is expected.
Can also use `toBeUndefined()`.